src/xrCore/Animation/Bone.hpp

```diff
@@ -465,41 +455,43 @@ class XRCORE_API CBoneData : public IBoneData
     using ChildFacesVec = xr_vector<FacesVec>;
     ChildFacesVec child_faces; // shared
 
-    CBoneData(u16 ID) : SelfID(ID) { VERIFY(SelfID != BI_NONE); }
-    virtual ~CBoneData() {}
+    explicit CBoneData(u16 ID) : SelfID(ID) { VERIFY(SelfID != BI_NONE); }
+    virtual ~CBoneData() = default;
 #ifdef DEBUG
     typedef svector<int, 128> BoneDebug;
     void DebugQuery(BoneDebug& L);
 #endif
     IC void SetParentID(u16 id) { ParentID = id; }
-    IC u16 GetSelfID() const { return SelfID; }
-    IC u16 GetParentID() const { return ParentID; }
+    [[nodiscard]] IC u16 GetSelfID() const override { return SelfID; }
+    [[nodiscard]] IC u16 GetParentID() const override { return ParentID; }
     // assign face
     void AppendFace(u16 child_idx, u16 idx) { child_faces[child_idx].push_back(idx); }
     // Calculation
     void CalculateM2B(const Fmatrix& Parent);
 
 private:
-    IBoneData& GetChild(u16 id) { return *children[id]; }
-    const IBoneData& GetChild(u16 id) const { return *children[id]; }
-    u16 GetNumChildren() const { return (u16)children.size(); }
-    const SJointIKData& get_IK_data() const { return IK_data; }
-    const Fmatrix& get_bind_transform() const { return bind_transform; }
-    const SBoneShape& get_shape() const { return shape; }
-    const Fobb& get_obb() const { return obb; }
-    const Fvector& get_center_of_mass() const { return center_of_mass; }
-    float get_mass() const { return mass; }
-    virtual u16 get_game_mtl_idx() const override { return game_mtl_idx; }
-    virtual shared_str GetMaterialName() const override { return name; }
-    float lo_limit(u8 k) const { return IK_data.limits[k].limit.x; }
-    float hi_limit(u8 k) const { return IK_data.limits[k].limit.y; }
+    IBoneData& GetChild(u16 id) override { return *children[id]; }
+    [[nodiscard]] const IBoneData& GetChild(u16 id) const override { return *children[id]; }
+    [[nodiscard]] u16 GetNumChildren() const override { return children.size(); }
+    [[nodiscard]] const SJointIKData& get_IK_data() const override { return IK_data; }
+    [[nodiscard]] const Fmatrix& get_bind_transform() const override { return bind_transform; }
+    [[nodiscard]] const SBoneShape& get_shape() const override { return shape; }
+    [[nodiscard]] const Fobb& get_obb() const override { return obb; }
+    [[nodiscard]] const Fvector& get_center_of_mass() const override { return center_of_mass; }
+    [[nodiscard]] float get_mass() const override { return mass; }
+    [[nodiscard]] u16 get_game_mtl_idx() const override { return game_mtl_idx; }
+    [[nodiscard]] shared_str GetMaterialName() const override { return name; }
+    [[nodiscard]] float lo_limit(u8 k) const override { return IK_data.limits[k].limit.x; }
+    [[nodiscard]] float hi_limit(u8 k) const override { return IK_data.limits[k].limit.y; }
 
 public:
-    virtual size_t mem_usage()
+    virtual std::size_t mem_usage()
```

Comment:
```suggestion
    virtual size_t mem_usage()
```

Reply:
![image](https://user-images.githubusercontent.com/22966523/100860972-401e3580-34a2-11eb-81d8-8459139e07e0.png)

Reply:
No, size_t is considered as built-in C++ type. Probably, we can workaround this in another way.

Reply:
I think it is not. Some `#include` should be added to fix this.

Reply:
Well, agree that it is `std` type, not built-in... But we use `size_t` instead of `std::size_t`, so any issues with that should be fixed in another way and in another pull request.

Reply:
Restored.